[Bug]: IndexNode panics if failed to create index

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

[2022/11/24 18:26:41.593 +08:00] [WARN] [indexcgowrapper/helper.go:76] ["failed to create index, C Runtime Exception: [UnexpectedError] Error in knowhere::VecIndexPtr knowhere::VecIndexFactory::CreateVecIndex(const IndexType&, knowhere::IndexMode) at VecIndexFactory.cpp:61: Invalid index type DISKANN\n"]
[2022/11/24 18:26:41.593 +08:00] [ERROR] [indexnode/task.go:366] ["failed to create index"] [error="[UnexpectedError] Error in knowhere::VecIndexPtr knowhere::VecIndexFactory::CreateVecIndex(const IndexType&, knowhere::IndexMode) at VecIndexFactory.cpp:61: Invalid index type DISKANN"] [stack="github.com/milvus-io/milvus/internal/indexnode.(*indexBuildTask).BuildDiskAnnIndex.func1\n\t/home/yah01/src/milvus/internal/indexnode/task.go:366\ngithub.com/milvus-io/milvus/internal/util/concurrency.(*Pool).Submit.func1\n\t/home/yah01/src/milvus/internal/util/concurrency/pool.go:46\ngithub.com/panjf2000/ants/v2.(*goWorker).run.func1\n\t/home/yah01/go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/worker.go:70"]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2910174]

goroutine 2303 [running]:
github.com/milvus-io/milvus/internal/util/indexcgowrapper.(*CgoIndex).CleanLocalData.func1(0xc000bd7358?)
	/home/yah01/src/milvus/internal/util/indexcgowrapper/index.go:339 +0x14
github.com/milvus-io/milvus/internal/util/indexcgowrapper.(*CgoIndex).CleanLocalData(0xc00108e928?)
	/home/yah01/src/milvus/internal/util/indexcgowrapper/index.go:339 +0x19
github.com/milvus-io/milvus/internal/indexnode.(*indexBuildTask).BuildDiskAnnIndex(0xc001077080, {0x4089b50?, 0xc000c6ec40})
	/home/yah01/src/milvus/internal/indexnode/task.go:377 +0x1114
github.com/milvus-io/milvus/internal/indexnode.(*indexBuildTask).BuildIndex(0xc001077080, {0x4089b50, 0xc000c6ec40})
	/home/yah01/src/milvus/internal/indexnode/task.go:241 +0xa5
github.com/milvus-io/milvus/internal/indexnode.(*TaskScheduler).processTask.func1(0xc000bd7e00)
	/home/yah01/src/milvus/internal/indexnode/task_scheduler.go:205 +0x82
github.com/milvus-io/milvus/internal/indexnode.(*TaskScheduler).processTask(0xc000cbdb40, {0x409b720, 0xc001077080}, {0xc000399520?, 0xc000b34c60?})
	/home/yah01/src/milvus/internal/indexnode/task_scheduler.go:218 +0x409
github.com/milvus-io/milvus/internal/indexnode.(*TaskScheduler).indexBuildLoop.func1(0x18f9d20?, {0x409b720?, 0xc001077080?})
	/home/yah01/src/milvus/internal/indexnode/task_scheduler.go:247 +0x6c
created by github.com/milvus-io/milvus/internal/indexnode.(*TaskScheduler).indexBuildLoop
	/home/yah01/src/milvus/internal/indexnode/task_scheduler.go:245 +0x298

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/internal/indexnode/task.go
+++ b/internal/indexnode/task.go
@@ -379,7 +379,7 @@ func (it *indexBuildTask) BuildDiskAnnIndex(ctx context.Context) error {
 			return nil, nil
 		}).Await()
 		if err != nil {
-			if it.index.CleanLocalData() != nil {
+			if it.index != nil && it.index.CleanLocalData() != nil {
 				log.Ctx(ctx).Error("failed to clean cached data on disk after build index failed",
 					zap.Int64("buildID", it.BuildID),
 					zap.Int64("index version", it.req.GetIndexVersion()))

